fem/coefficient.cpp

```diff
@@ -277,6 +277,30 @@ void VectorRestrictedCoefficient::Eval(
    }
 }
 
+void VectorFunctionRestrictedCoefficient::Eval(Vector &V, ElementTransformation &T,
+                                               const IntegrationPoint &ip)
+{
+   double x[3];
+   Vector transip(x, 3);
+   
+   T.Transform(ip, transip);
+
+   V.SetSize(vdim);
+   if (active_attr[T.Attribute-1])
```

Comment:
Why the -1 here.  Are the indices 1-based in your TDFunction definition?  FORTRAN?

Reply:
This function appears to used only when our boundary conditions are applied. It looks like when examining ```BCManager.hpp``` and ```BCManager.hpp``` that the attribute id being used is 1-based. @srwopschall  wrote a majority of the boundary condition code, so he might have better input here for why it's attribute id is 1-based instead of 0-based.